credentials/alts/utils_test.go

```diff
@@ -147,14 +155,61 @@ func TestAuthInfoFromPeer(t *testing.T) {
 	}
 }
 
-type fakeALTSAuthInfo struct{}
+func TestClientAuthorizationCheck(t *testing.T) {
+	ctx := context.Background()
+	altsAuthInfo := &fakeALTSAuthInfo{testServiceAccount1}
+	p := &peer.Peer{
+		AuthInfo: altsAuthInfo,
+	}
+	for _, tc := range []struct {
+		desc                    string
+		ctx                     context.Context
+		expectedServiceAccounts []string
+		success                 bool
+		code                    codes.Code
+	}{
+		{
+			"working case",
+			peer.NewContext(ctx, p),
+			[]string{testServiceAccount1, testServiceAccount2},
+			true,
+			codes.OK, // err is nil, code is OK.
+		},
+		{
+			"context does not have AuthInfo",
+			ctx,
+			[]string{testServiceAccount1, testServiceAccount2},
+			false,
+			codes.Unknown, // err doesn't have code, code is Unknown.
```

Comment:
If there is no AuthInfo from context, can we return NotFound instead of Unknown?

Reply:
We don't return `Unknown`. We return whatever error `AuthInfoFromContext` returns. And since that error does not have a code, `status.Code()` (below) treats it as an `Unknown` code.

I can change the code to wrap `AuthInfoFromContext` with a `NotFound` error. WDYT?

Reply:
SG with NotFound error.

Reply:
Done.